tests/client/test_properties.py

```diff
@@ -49,3 +49,12 @@ def test_client_timeout():
     assert client.timeout.read == expected_timeout
     assert client.timeout.write == expected_timeout
     assert client.timeout.pool == expected_timeout
+
+
+def test_client_event_hooks():
+    def on_request(request):
+        pass  # pragma: nocover
+
+    client = Client()
+    client.event_hooks = {"request": on_request}  # type: ignore
```

Comment:
Another reason why I'm advocating for a straightforward "dict of lists of callables" is that we wouldn't need any `# type: ignore`…

```suggestion
    client.event_hooks = {"request": [on_request]}
```